pkg/kubectl/resource_printer.go

```diff
@@ -145,6 +145,9 @@ type ResourcePrinter interface {
 	// Print receives a runtime object, formats it and prints it to a writer.
 	PrintObj(runtime.Object, io.Writer) error
 	HandledResources() []string
+	//Can be used to print out warning/clarifications if needed
+	//after all objects were printed
+	FinishPrint(io.Writer) error
```

Comment:
`PrintObj` is what calls `printPodList`, so if you put the warning message in there before returning (maybe use a defer?), it should give you the same output as
```
err := printer.PrintObj(e.Object, out)
if err == nil {
		printer.FinishPrint(out)
}
```
which is what you're doing now.